src/Rules/DisallowedConstructs/DisallowedWeakComparisonRule.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\DisallowedConstructs;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp;
+use PhpParser\Node\Expr\BinaryOp\Equal;
+use PhpParser\Node\Expr\BinaryOp\NotEqual;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+
+/**
+ * @implements Rule<BinaryOp>
+ */
+class DisallowedWeakComparisonRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return BinaryOp::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if ($node instanceof Equal || $node instanceof NotEqual) {
+			return [
+				'Weak comparison via "' . $node->getOperatorSigil() . '" is not allowed. Use strong comparison instead.',
```

Comment:
The correct nomenclature isn't weak/strong, but loose/strict.

Also please use the correct alternative in the error message. `==` should have `===` in the second part of the message, `!=` should have `!==` in the second part of the message.

Reply:
Also: Please use RuleErrorBuilder here.

Reply:
Also: Please use `sprintf` instead of string concatenation.

Reply:
Should be fixed by https://github.com/phpstan/phpstan-strict-rules/pull/177/commits/56a7475d7185c829770fc9d1e721c5a3aaadceb1. I only took the others rules in the directory as examples which don't use `RuleErrorBuilder` either.

Reply:
sorry, I noticed that the class / rule name is still using the other terms. does it make sense to adapt them too for consistency?

Reply:
Yes please :)

Reply:
Nicely spotted. Fixed that, too.